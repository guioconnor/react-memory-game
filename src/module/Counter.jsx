import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  min-width: 50px;
  text-align: center;
`;

const Counter = ({ counter }) => <Container>{counter}</Container>;

export default Counter;
