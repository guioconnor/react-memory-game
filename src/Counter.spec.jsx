import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Counter from './Counter';

describe('<Counter />', () => {
  describe('when `counter` is undefined', () => {
    const container = shallow(<Counter />);
    it('should return an empty placeholder ', () => {
      expect(toJson(container)).toMatchSnapshot('undefinedCounter');
    });
  });

  describe('when a valid `counter is passed`', () => {
    const container = shallow(<Counter counter={42} />);
    it('should render the passed value as a child of the counter', () => {
      expect(toJson(container)).toMatchSnapshot('undefinedCounter');
    });
  });
});
