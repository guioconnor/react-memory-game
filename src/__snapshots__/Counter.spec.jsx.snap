// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> when \`counter\` is undefined should return an empty placeholder : undefinedCounter 1`] = `
<div
  className="memory-game--counter"
/>
`;

exports[`<Counter /> when a valid \`counter is passed\` should render the passed value as a child of the counter: undefinedCounter 1`] = `
<div
  className="memory-game--counter"
>
  42
</div>
`;
